DiscordChatExporter.Core/Exporting/Filtering/MessageFilter.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using DiscordChatExporter.Core.Discord.Data;
+using DiscordChatExporter.Core.Exporting.Filtering.Parsing;
+using Superpower;
+using Superpower.Model;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public abstract partial class MessageFilter
+    {
+        public abstract bool Filter(Message message);
+    }
+
+    public partial class MessageFilter
+    {
+        public static MessageFilter CreateFilter(string key, string value)
+        {
+            return key.ToLower() switch
+            {
+                "contains" => new ContainsMessageFilter(value),
+                "from" => new FromMessageFilter(value),
+                "has" => new HasMessageFilter(value),
+                "mentions" => new MentionsMessageFilter(value),
+                _ => throw new ArgumentException($"Invalid filter type '{key}'.", "key")
+            };
+        }
+
+        public static MessageFilter Parse(string value, IFormatProvider? formatProvider = null) =>
+            FilterParser.TryParse(value, out var filter, out var error, out var position) ? filter : throw new ParseException(error, position);
```

Comment:
I couldn't actually get these errors to show up in the console. They were being replaced with a generic message about being unable to set the `--filter` flag. Is CliFx masking them?

Reply:
I believe it should still include the original error message:

https://github.com/Tyrrrz/CliFx/blob/df73a0bfe80e4984e13a5f680cc5cd0382eead2e/CliFx/CommandBinder.cs#L164-L171

What is the error message that gets printed?

Reply:
It looks like a `TargetInvocationException` with the message "Error: Exception has been thrown by the target of an invocation." I did some investigating and it seems that the reflection CliFx uses is wrapping the thrown exception because of the call to .Invoke(), with the original exception being stored as an InnerException. Setting an invalid partition produces the same behaviour.

Reply:
Uh oh. Leave it like this for now, I will look at it separately. Thanks.